lighthouse-core/audits/seo/hreflang.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * @license Copyright 2017 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const Audit = require('../audit');
+const LinkHeader = require('http-link-header');
+const axeCore = require('axe-core');
```

Comment:
turns out I can't import 'axe-core' as it wasn't imported anywhere else before and since it's huge, it blows our bundle budget 🤔

Reply:
any chance we can just import the languages file?

`const validLangs = require('axe-core/lib/commons/utils/valid-langs.js')`

Reply:
meta observation: there's only ~17k possible 3-character codes and ~8k of them are valid

how much value is the list really providing over a `/^[a-z]{2,3}$/` regex? are mistakes likely to result in a different invalid 3-character code?

Reply:
Didn't realize `require` can import single files that are not in the root folder of the package, thanks! I had to use a slight hack (global variable) to get things working though.

45% of 3-character codes and 25% of 2-character codes are valid - I agree that it's a lot, but IMO it still leaves a huge margin for people to make "catchable" mistakes. We may also extend this audit with region and script codes validation in the future. @rviscomi WDYT?

Reply:
Not a fan of regex pattern matching as a substitute for whitelist checking. Too many false positives.

Reply:
One option is putting it in a different data structure when browserifying. A quick test of a pretty naive trie got it down to about 1/2 the size raw, 1/5 the size after gzip (19915 -> 3751). Being more clever can get the trie down to about 1/4 the size of the current file), but the gzip size doesn't really budge after that first drop.

Probably not worth pursuing at this point, but maybe worth leaving a note for when we go back looking for the low hanging fruit of shrinking bundle size.